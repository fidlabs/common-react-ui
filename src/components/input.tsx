import type { HTMLAttributes } from "react";

export type InputProps = HTMLAttributes<HTMLInputElement>;

export function Input(props: InputProps) {
  return <input {...props} />;
}
