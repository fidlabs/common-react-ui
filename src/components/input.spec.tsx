import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { Input } from "./input";

describe("Input component", () => {
  it("renders correctly", () => {
    const { container } = render(<Input placeholder="Email" />);
    expect(container).toMatchSnapshot();
  });
});
