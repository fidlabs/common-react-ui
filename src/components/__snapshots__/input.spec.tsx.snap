// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input component > renders correctly 1`] = `
<div>
  <input
    class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-dodger-blue focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50"
    placeholder="Email"
  />
</div>
`;
